plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx

```diff
@@ -75,30 +69,46 @@ const withSidebarNotices = createHigherOrderComponent(
 						? clientId
 						: parents[ targetParentBlock ],
 			};
-		} );
+		},
+		[ clientId ]
+	);
+
+	return (
+		( isCart || isCheckout ) && (
+			<InspectorControls>
+				<IncompatibleExtensionsNotice
+					block={
+						isCart ? 'woocommerce/cart' : 'woocommerce/checkout'
+					}
+					clientId={ parentId }
+				/>
 
-		// Show sidebar notices only when a WooCommerce block is selected.
-		if (
-			! blockName.startsWith( 'woocommerce/' ) ||
-			! isBlockSelected ||
-			! ( isCart || isCheckout )
-		) {
-			return <BlockEdit key="edit" { ...props } />;
-		}
+				<DefaultNotice block={ isCheckout ? 'checkout' : 'cart' } />
+				<CartCheckoutFeedbackPrompt />
+			</InspectorControls>
+		)
+	);
+};
+
+const withSidebarNotices = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		const {
+			clientId,
+			name: blockName,
+			isSelected: isBlockSelected,
+		} = props;
 
 		return (
 			<>
-				<InspectorControls>
-					<IncompatibleExtensionsNotice
-						block={
-							isCart ? 'woocommerce/cart' : 'woocommerce/checkout'
-						}
-						clientId={ parentId }
-					/>
-
-					<DefaultNotice block={ isCheckout ? 'checkout' : 'cart' } />
-					<CartCheckoutFeedbackPrompt />
-				</InspectorControls>
+				{
+					// Show sidebar notices only when a WooCommerce block is selected.
+					// This early check helps prevent expensive and unnecessary work
+					// in the block editor store.
+					blockName.startsWith( 'woocommerce/' ) &&
```

Comment:
Do we really need this for _all_ woo commerce blocks?

Reply:
No, as far as I know, we're really only interested in the cart and checkout at the moment.

@senadir has a better approach in mind, that will avoid the need to connect to the store altogether. It might need some product input, however, so the plan is to use this approach for now, and replace it later if possible.